### ISSUE TITLE
How can I add virtualenv --always-copy options?

### ISSUE THREAD

**Issue by pytoxbot** [tox-dev/tox#218]
- Bitbucket: https://bitbucket.org/hpk42/tox/issue/218
- Originally reported by: [pahaz](http://bitbucket.org/pahaz)
- Originally created at: 2015-02-19T18:57:23.391

**Comment by obestwalter** [comment]
Closing this as it is very old and things have moved a lot since then.

**Comment by gaborbernat** [comment]
I've bumped into this today, apparently this is needed so that the vagrant file you added can run on Windows 😲 we should probably reopen it to make this happen for ourselves (development purpose)

**Comment by gaborbernat** [comment]
@obestwalter

**Comment by gaborbernat** [comment]
The need of this always copy is optional on the invoking environment, so this should be either a tox cli argument, or taken from the env; what do you guys think?

```bash
[vagrant@vagrant-arch vagrant]$ /home/vagrant/.pyenv/versions/3.6.0/bin/python3.6 -m virtualenv --python /home/vagrant/.pyenv/shims/python2.7  py27  
Running virtualenv with interpreter /home/vagrant/.pyenv/shims/python2.7
New python executable in /vagrant/py27/bin/python2.7
Also creating executable in /vagrant/py27/bin/python
Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/3.6.0/lib/python3.6/site-packages/virtualenv.py", line 2328, in <module>
    main()
  File "/home/vagrant/.pyenv/versions/3.6.0/lib/python3.6/site-packages/virtualenv.py", line 713, in main
    symlink=options.symlink)
  File "/home/vagrant/.pyenv/versions/3.6.0/lib/python3.6/site-packages/virtualenv.py", line 925, in create_environment
    site_packages=site_packages, clear=clear, symlink=symlink))
  File "/home/vagrant/.pyenv/versions/3.6.0/lib/python3.6/site-packages/virtualenv.py", line 1370, in install_python
    os.symlink(py_executable_base, full_pth)
OSError: [Errno 71] Protocol error
[vagrant@vagrant-arch vagrant]$ /home/vagrant/.pyenv/versions/3.6.0/bin/python3.6 -m virtualenv --python /home/vagrant/.pyenv/shims/python2.7  py27 --always-copy
Running virtualenv with interpreter /home/vagrant/.pyenv/shims/python2.7
New python executable in /vagrant/py27/bin/python2.7
Also creating executable in /vagrant/py27/bin/python
Installing setuptools, pip, wheel...done.
[vagrant@vagrant-arch vagrant]$ 
```

**Comment by obestwalter** [comment]
@gaborbernat thanks for trying this. Let's reopen this then.

I have no opinion about from where to fetch it yet. Is there some "prior art" in the code, where we could take inspiration from for this kind of problem?

**Comment by obestwalter** [comment]
just realized that @asottile added this already as a cli arg for 2.6, so I guess we can close this as resolved?

http://tox.readthedocs.io/en/latest/example/basic.html?highlight=alwayscopy#prevent-symbolic-links-in-virtualenv

**Comment by asottile** [comment]
heh, I don't want to take credit for something I didn't do, here's the changeset thanks to @igordcard

https://github.com/tox-dev/tox/pull/434

**Comment by obestwalter** [comment]
ups - sorry, I had associated that with you for whatever reason - anyway you did other great things recently :)

Thanks to @igordcard then 🎉

**Comment by gaborbernat** [comment]
yeah, it works via that :+1:

**Comment by obestwalter** [comment]
fixed since 2.6 - see http://tox.readthedocs.io/en/latest/config.html?highlight=alwayscopy#confval-alwayscopy=True|False
